### PR TITLE
py-netcdf4: a couple of improvements

### PR DESCRIPTION
**Pull request by skosukhin**
- avoid recompilation
- make the older versions to work with `hdf5@1.10:`

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-netcdf4/package.py [hunk 1]
```diff
@@ -32,6 +32,13 @@ class PyNetcdf4(PythonPackage):
     # following patch disables the usage of pkg-config at all.
     patch('disable_pkgconf.patch')
 
+    # Older versions of the package get a false negative result when checking
+    # the version of HDF5.
+    patch('check_hdf5version.patch', when='@:1.2.9 ^hdf5@1.10:')
+
+    # We can skip the 'build' phase to avoid recompilation of the library.
+    phases = ['install']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-netcdf4/package.py:43]: I've noticed this as well, I'm thinking about switching the default `phases` to just install, but want to wait until we have better CI to try something like that.

**Review by adamjstewart — APPROVED** [review]
Would you have any interest in being an official "maintainer" for this package?

**Comment by skosukhin** [comment]
@adamjstewart I also took the liberty to add the latest version.

**Review by adamjstewart — APPROVED** [review]

**Comment by skosukhin** [comment]
BTW, once/if #8623 is merged, we can replace `patch('disable_pkgconf.patch')` with `depends_on('pkgconf', type='build')`.

**Pull request merged**